In [113]:
import pandas as pd
import plotly.express as px


In [114]:
df = pd.read_csv("final_table.csv")
df

Unnamed: 0  Age          Primary_streaming_service  Hours_per_day  \
0             0   25                        Apple Music              2   
1             1   16            Other streaming service              4   
2             2   22                            Spotify             10   
3             3   26                            Spotify              6   
4             4   18  I do not use a streaming service.              4   
..          ...  ...                                ...            ...   
731         731   29                      YouTube Music              2   
732         732   17                            Spotify              4   
733         733   17                            Spotify              3   
734         734   13                            Spotify              1   
735         735   24  I do not use a streaming service.              8   

     While_working  Instrumentalist  Composer         Fav_genre  Exploratory  \
0            False            False     False               EDM        False   
1             True            False     False               EDM         True   
2            False             True      True               EDM        False   
3             True            False     False               EDM        False   
4             True             True     False               EDM        False   
..             ...              ...       ...               ...          ...   
731           True            False     False           Hip hop         True   
732          False             True     False         Classical         True   
733           True            False     False         Classical         True   
734           True            False     False  Video game music         True   
735           True            False      True  Video game music         True   

     Foreign_languages  ... Anxiety Depression Insomnia OCD Music_effects  \
0                 True  ...      10          3        1   4       Improve   
1                 True  ...       4          7        1   3       Improve   
2                 True  ...       4          7        7   3       Improve   
3                 True  ...      10         10        8   2     No effect   
4                 True  ...       8          7       10   6       Improve   
..                 ...  ...     ...        ...      ...  ..           ...   
731               True  ...       2          2        2   5       Improve   
732               True  ...       7          9        2   8        Worsen   
733               True  ...       6          6        8   2       Improve   
734              False  ...      10         10        8   4        Worsen   
735              False  ...       6          4        8   8       Improve   

           Date      Time  Year Month Day  
0    2022-08-28  23:26:58  2022     8  28  
1    2022-09-13  01:55:43  2022     9  13  
2    2022-08-28  17:15:18  2022     8  28  
3    2022-08-30  05:16:13  2022     8  30  
4    2022-09-02  09:25:05  2022     9   2  
..          ...       ...   ...   ...  ..  
731  2022-11-09  01:55:20  2022    11   9  
732  2022-08-28  23:23:50  2022     8  28  
733  2022-08-28  11:58:48  2022     8  28  
734  2022-10-22  15:20:29  2022    10  22  
735  2022-08-29  01:22:44  2022     8  29  

[736 rows x 37 columns]

**Stats**
Calculate stats for age, hours per day listening to music, frequency of different genres, anxiety levels, depression levels, insomnia levels and OCD **levels**

**Ages that listen to most hours per day and their results**

In [115]:
df2 =round(df[['Age',"Hours_per_day","Anxiety","Depression","Insomnia","OCD"]][(df["Age"] >= 18)&(df["Age"] <= 62)&(df["Age"] != 55)&(df["Age"] != 54)].groupby("Age",as_index=False).mean(),2)
df2

Age  Hours_per_day  Anxiety  Depression  Insomnia   OCD
0    18           3.56     5.67        4.35      3.80  2.75
1    19           4.18     5.56        4.23      2.70  1.89
2    20           3.08     6.12        5.10      3.80  2.45
3    21           3.04     6.04        5.36      3.42  3.17
4    22           3.54     6.49        6.03      3.51  2.82
5    23           3.59     6.38        5.54      3.81  3.05
6    24           4.50     6.35        6.00      4.30  3.65
7    25           2.55     6.68        5.23      3.09  2.18
8    26           3.55     5.73        5.59      3.27  2.23
9    27           5.00     5.28        5.17      4.17  2.39
10   28           2.69     5.92        5.69      3.08  1.92
11   29           3.00     7.15        5.08      4.23  3.38
12   30           3.64     7.64        5.91      6.09  4.45
13   31           3.38     7.08        5.92      5.38  3.77
14   32           3.62     6.00        4.94      3.62  3.06
15   33           2.62     5.50        5.25      4.00  2.25
16   34           1.88     4.38        6.12      4.12  1.25
17   35           3.14     5.86        3.57      2.71  2.29
18   36           3.14     5.86        4.29      3.71  1.14
19   37           1.80     7.00        5.60      7.60  3.80
20   38           2.33     4.33        4.83      5.83  3.67
21   39           1.00     6.00        4.00      1.00  1.00
22   40           3.80     3.80        3.40      3.40  1.20
23   41           3.25     8.00        5.25      4.50  2.25
24   42           1.33     6.83        5.33      6.17  3.33
25   43           2.00     3.50        2.50      2.75  1.00
26   44           2.00     8.00        7.00      6.33  2.33
27   46           1.00     0.00        0.00      1.00  0.00
28   48           3.00     6.50        5.50      0.00  2.00
29   49           2.40     3.80        4.60      4.80  0.80
30   50           1.00     0.00        0.00      7.00  0.00
31   51           5.00     1.00        2.00      1.00  0.00
32   53           4.00     1.00        0.67      0.67  0.33
33   56           3.80     3.40        2.40      4.40  5.00
34   57           2.00     3.33        2.00      5.67  0.00
35   58           1.67     3.00        4.00      4.33  1.00
36   59           1.50     1.00        0.50      1.50  0.50
37   60           1.67     3.83        2.50      3.17  0.67
38   61           2.00     5.50        4.50      2.50  1.50

In [116]:
px.line(x=df2['Age'],y=df2['OCD'])

It is important to notice, that when the average hours per day that people listen to music tends to lower its levels as the people gets older.
The levels of anxiety tends to be pretty high on ages from 20 to 30, dropping its level after 45 years old, similar to Depression
The levels of insomnia tends to be higher after 35
OCD levels are being reduced as the age increases, but after 50 the general levels of OCD are higher.

In [117]:
df_3 = round(df[["Age","Hours_per_day","Anxiety","Depression","Insomnia","OCD"]].corr(),1)
df_3

Age  Hours_per_day  Anxiety  Depression  Insomnia  OCD
Age            1.0           -0.0     -0.2        -0.1       0.0 -0.1
Hours_per_day -0.0            1.0      0.0         0.1       0.1  0.1
Anxiety       -0.2            0.0      1.0         0.5       0.3  0.3
Depression    -0.1            0.1      0.5         1.0       0.4  0.2
Insomnia       0.0            0.1      0.3         0.4       1.0  0.2
OCD           -0.1            0.1      0.3         0.2       0.2  1.0

The strongest correlation is found between anxiety and depression, that means
that people that feels anxiety are more likely to feel depressed over time.
The hours per day that people listen to music does not seems to have a relevant effect on peoples perception of Anxiety, Depression, Insomnia or OCD.

In [118]:
df_4 = round(df[["Fav_genre",'Age',"Hours_per_day","Anxiety","Depression","Insomnia","OCD"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Fav_genre").mean().sort_values(by=['Anxiety'],ascending=False).head(20),1)
df_4

Age  Hours_per_day  Anxiety  Depression  Insomnia  OCD
Fav_genre                                                                
Lofi              24.9            3.9      6.9         7.9       6.4  4.1
Latin             18.0            4.0      6.5         3.0       0.5  0.0
Folk              27.8            3.3      6.4         5.2       3.9  2.1
Rock              29.0            3.4      6.2         5.4       3.8  2.5
Pop               25.0            2.9      6.1         4.7       3.3  2.7
Jazz              27.2            5.5      6.1         4.8       3.9  2.8
K pop             19.7            3.7      6.1         4.1       3.5  2.1
Hip hop           25.5            3.2      5.9         5.7       3.3  2.9
Video game music  24.1            3.0      5.9         4.8       4.1  2.4
Metal             26.4            3.5      5.6         5.0       4.4  2.3
Rap               20.1            2.8      5.4         4.3       3.0  3.4
EDM               23.5            4.4      5.4         5.2       4.2  3.1
Country           24.6            3.4      5.2         4.5       2.8  2.9
R&B               23.8            3.4      4.9         3.5       2.5  2.9
Classical         28.9            2.9      4.6         4.1       3.4  2.2
Gospel            48.8            3.2      4.5         1.5       5.5  0.5

The people that reports Lofi as their fav_genre tends to have higher levels of Anxiety and Depression, also the average age of this people is 24.8 years old, that supports the previous analysis on the levels of anxiety and depression on the mid 20s.

In [119]:
df_5 = round(df[["Fav_genre","Hours_per_day","Anxiety"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Fav_genre").mean().sort_values(by=['Anxiety'],ascending=False).head(20),2).corr()
df_5

Hours_per_day   Anxiety
Hours_per_day       1.000000  0.307866
Anxiety             0.307866  1.000000

Analyzing the data from ages between 18 and 62 years old, we can see that the correlation between hours_per_day and anxiety tends to increase its value.

In [120]:
df_6 = round(df[["Frequency_Lofi","Age","Hours_per_day","Anxiety","Depression","Insomnia","OCD"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Frequency_Lofi").mean(),2)
df_6

Age  Hours_per_day  Anxiety  Depression  Insomnia   OCD
Frequency_Lofi                                                            
Never            29.13           3.11     5.49        4.60      3.51  2.43
Rarely           25.17           3.19     5.75        5.06      3.70  2.56
Sometimes        23.90           3.78     6.62        5.35      3.88  2.90
Very frequently  24.02           4.13     5.57        4.78      4.17  2.54

Analyzing the worst 3 genres in general in terms of levels of depression, anxiety, insomnia, OCD, etc. The levels of Anxiety tends to be higher when the people listen to this genre "sometimes", that means 3.78 hours a day.

In [121]:
df_7 = round(df[["Frequency_Rap","Age","Hours_per_day","Anxiety","Depression","Insomnia","OCD"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Frequency_Rap").mean(),2)
df_7

Age  Hours_per_day  Anxiety  Depression  Insomnia   OCD
Frequency_Rap                                                             
Never            28.90           2.95     5.26        3.96      3.21  2.32
Rarely           26.50           3.30     6.24        5.46      4.22  2.76
Sometimes        25.64           3.59     5.82        5.10      3.62  2.48
Very frequently  22.79           3.96     5.94        5.09      3.77  2.84

In the case of Rap, the people that presents the higher levels of anxiety listen to them rarely, that means less than 3.3 hours a day.

In [122]:
df_8 =round(df[["Frequency_EDM","Age","Hours_per_day","Anxiety","Depression","Insomnia","OCD"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Frequency_EDM").mean(),1)
df_8

Age  Hours_per_day  Anxiety  Depression  Insomnia  OCD
Frequency_EDM                                                           
Never            28.0            3.0      5.5         4.4       3.4  2.3
Rarely           24.6            3.3      6.2         5.5       3.8  2.6
Sometimes        25.8            3.8      6.0         5.0       3.7  2.7
Very frequently  25.5            4.1      5.8         5.1       4.4  3.2

In the case of EDM it presents the highest levels of insomnia for people that listen to it very frequently, that means more than 4.1 hours a day.

**Perception Analysis** Music effect analysis to understand how participant perceive the impact of music on their well being

In [123]:
df_9 = round(df[["Music_effects","Age","Hours_per_day","OCD","Insomnia","Depression","Anxiety"]][(df["Age"] >= 18)&(df["Age"] <= 62)].groupby("Music_effects").mean(),2)
df_9

Age  Hours_per_day   OCD  Insomnia  Depression  Anxiety
Music_effects                                                           
Improve        25.91           3.50  2.70      3.73        5.00     6.03
No effect      27.60           3.23  2.25      3.65        4.47     5.18
Not specified  19.67           2.00  2.00      2.50        4.83     4.67
Worsen         27.09           2.55  2.91      5.00        7.45     7.09

In [124]:
from pathlib import Path
filepath = Path('/content/sample_data/age_analysis.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df2.to_csv(filepath)

In [125]:
from pathlib import Path
filepath = Path('/content/sample_data/corr_hours.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_3.to_csv(filepath)

In [126]:
from pathlib import Path
filepath = Path('/content/sample_data/genre_analysis.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_4.to_csv(filepath)

In [127]:
from pathlib import Path
filepath = Path('/content/sample_data/genre_corr.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_5.to_csv(filepath)

In [128]:
from pathlib import Path
filepath = Path('/content/sample_data/lofi.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_6.to_csv(filepath)

In [129]:
from pathlib import Path
filepath = Path('/content/sample_data/rap.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_7.to_csv(filepath)

In [130]:
from pathlib import Path
filepath = Path('/content/sample_data/edm.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_8.to_csv(filepath)

In [131]:
from pathlib import Path
filepath = Path('/content/sample_data/perception_analysis.csv')
filepath.parent.mkdir(parents=True, exist_ok=True)
df_9.to_csv(filepath)

**Correlation Analysis**
Correlation analysis with pearson or spearman using the moods columns

**texto en negrita**